app/authorization.go

```diff
@@ -32,12 +32,12 @@ func (a *App) SessionHasPermissionToTeam(session model.Session, teamId string, p
 	return a.RolesGrantPermission(session.GetUserRoles(), permission.Id)
 }
 
-func (a *App) SessionHasPermissionToChannel(session model.Session, channelId string, permission *model.Permission) bool {
+func (a *App) SessionHasPermissionToChannel(session model.Session, channelId string, permission *model.Permission, includeDeleted bool) bool {
```

Comment:
This makes me feel quite uneasy. Either you have permission to a channel, or you don't. Is there a (reasonable) way we can have the archived/not-archived checking outside of the permissions handler? Does this relate to your general comment about the config setting?

Reply:
Yeah I would also like to hear the argument for permissions being concerned with archive/not-archive state because my default would be to say it's not a permissions concern.